src/main/kotlin/app/revanced/patches/youtube/misc/settings/bytecode/patch/SettingsPatch.kt

```diff
@@ -39,38 +40,55 @@ import java.io.Closeable
 @SettingsCompatibility
 @Version("0.0.1")
 class SettingsPatch : BytecodePatch(
-    listOf(LicenseActivityFingerprint, ReVancedSettingsActivityFingerprint, ThemeSetterFingerprint)
+    listOf(LicenseActivityFingerprint, ReVancedSettingsActivityFingerprint, ThemeSetterSystemFingerprint, ThemeSetterAppFingerprint)
 ) {
     override fun execute(context: BytecodeContext): PatchResult {
         val licenseActivityResult = LicenseActivityFingerprint.result!!
         val settingsResult = ReVancedSettingsActivityFingerprint.result!!
-        val themeSetterResult = ThemeSetterFingerprint.result!!
+        val themeSetterSystemResult = ThemeSetterSystemFingerprint.result!!
+        val themeSetterAppResult = ThemeSetterAppFingerprint.result!!
 
         val licenseActivityClass = licenseActivityResult.mutableClass
         val settingsClass = settingsResult.mutableClass
 
         val onCreate = licenseActivityResult.mutableMethod
         val setThemeMethodName = "setTheme"
         val initializeSettings = settingsResult.mutableMethod
+        val themeSetterAppMethod = themeSetterAppResult.mutableMethod
 
-        val setThemeInstruction =
+        val setSystemThemeInstruction =
             "invoke-static {v0}, Lapp/revanced/integrations/utils/ThemeHelper;->setTheme(Ljava/lang/Object;)V".toInstruction(
-                themeSetterResult.mutableMethod
+                themeSetterSystemResult.mutableMethod
             )
 
-        // add instructions to set the theme of the settings activity
-        themeSetterResult.mutableMethod.implementation!!.let {
+        // add instructions to set the theme of the settings activity, based on Android system tint
+        themeSetterSystemResult.mutableMethod.implementation!!.let {
             it.addInstruction(
-                themeSetterResult.scanResult.patternScanResult!!.startIndex,
-                setThemeInstruction
+                themeSetterSystemResult.scanResult.patternScanResult!!.startIndex,
+                setSystemThemeInstruction
             )
 
             it.addInstruction(
                 it.instructions.size - 1, // add before return
-                setThemeInstruction
+                setSystemThemeInstruction
             )
         }
 
+        fun setAppThemeInstructions(value: Int): String {
+            return """
+                const/4 v0, 0x$value
+                invoke-static {v0}, Lapp/revanced/integrations/utils/ThemeHelper;->setTheme(I)V
+            """
+        }
```

Comment:
```suggestion
        fun setAppThemeInstructions(value: Int) = """
                const/4 v0, 0x$value
                invoke-static {v0}, Lapp/revanced/integrations/utils/ThemeHelper;->setTheme(I)V
            """
```

Make sure to format the code.